test/rekt/resources/apiserversource/apiserversource.go

```diff
@@ -166,3 +171,94 @@ func labelSelectorToStringMap(selector *metav1.LabelSelector) map[string]interfa
 
 	return r
 }
+
+func MatchLabels(labels1 map[string]string, labels2 map[string]string) bool {
+	for k, v := range labels1 {
+		if labels2[k] != v {
+			return false
+		}
+	}
+	return true
+}
+
+func TestLabels() map[string]string {
+	return map[string]string{
+		"testkey":  "testvalue",
+		"testkey1": "testvalue1",
+		"testkey2": "testvalue2",
+	}
+}
+
+func VerifyNodeSelectorDeployment(source string) feature.StepFn {
+	return func(ctx context.Context, t feature.T) {
+		env := environment.FromContext(ctx)
+		ns := env.Namespace()
+
+		kubeClient := client.Get(ctx)
+
+		deps, err := kubeClient.AppsV1().Deployments(ns).List(ctx, metav1.ListOptions{})
+		if err != nil {
+			t.Fatalf("error getting deployment: %v", err)
+		}
+
+		var dep appsv1.Deployment
+
+		for _, d := range deps.Items {
+			if kmeta.ChildName(fmt.Sprintf("apiserversource-%s-", source), string(d.GetUID())) == d.Name {
+				dep = d
+			}
+		}
+
+		if !MatchLabels(dep.Spec.Template.Spec.NodeSelector, TestLabels()) {
+			t.Fatalf("NodeSelector labels do not match: %v", dep.Spec.Template.Spec.NodeSelector)
+		}
+	}
+}
+
+func SetupNodeLabels() feature.StepFn {
+	return func(ctx context.Context, t feature.T) {
+		nodes, err := client.Get(ctx).CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+		if err != nil {
+			t.Fatalf("Could not list nodes: %v", err)
+		}
+
+		if len(nodes.Items) == 0 {
+			t.Fatal("No nodes found")
+		}
+
+		node := &nodes.Items[0]
+
+		node.Labels["testkey"] = "testvalue"
+		node.Labels["testkey1"] = "testvalue1"
+		node.Labels["testkey2"] = "testvalue2"
+
+		_, err = client.Get(ctx).CoreV1().Nodes().Update(ctx, node, metav1.UpdateOptions{})
+
+		if err != nil {
+			t.Fatalf("Could not update node: %v", err)
+		}
+	}
+}
+
+func ResetNodeLabels(ctx context.Context, t feature.T) {
+	nodes, err := client.Get(ctx).CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		t.Fatalf("Could not list nodes: %v", err)
+	}
+
+	if len(nodes.Items) == 0 {
+		t.Fatal("No nodes found")
+	}
```

Comment:
Can you select by the labels in the `metav1.ListOptions{}`?

Otherwise, we can't know the ordering you get here for sure, so just selecting the first item may not work.

Reply:
actually any node we select might not matter right ? we just want a node to patch those labels so we have a valid nodeselector . correct ?

Reply:
Not in this case, as we are trying to remove the labels, not add them

Reply:
oh ya good catch . appreciate it. ill fix that